### PR TITLE
JDBC Support

### PR DESCRIPTION
**Pull request by cgivre**
This PR gives the user the choice of using the REST interface OR the JDBC interface to query Drill.  
What remains to be done:

1.  At the moment, the location for the Drill driver is hard coded.  We obviously need to fix that and prompt the user to D/L the JDBC driver from MapR's repo.

### CODE REVIEW

File: sqlalchemy_drill/base.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+    'varchar': types.String,
+    'VARCHAR': types.String,
+    'character varying': types.String,
+    'CHARACTER VARYING': types.String,
+    'ANY': types.String,
+    'any': types.String,
+    'map': types.UserDefinedType,
+    'MAP': types.UserDefinedType,
+    'list': types.UserDefinedType,
+    'LIST': types.UserDefinedType
```
Comment (villebro) [inline sqlalchemy_drill/base.py]: It seems excessive to have both upper and lower case variants here, especially as calls to this map seem to always be done using `.lower()`.

File: requirements/common.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 sqlalchemy
 pydrill
+JayDeBeApi
```
Comment (villebro) [inline requirements/common.txt:3]: Missing line change.

File: sqlalchemy_drill/jdbc.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        return "jdbc:drill:drillbit=%s:%s" % (
+            url.host,
+            url.port or 31010
+        )
+
+    @classmethod
+    def dbapi(cls):
+        return jaydebeapi
+
+dialect = DrillDialect_jdbc
```
Comment (villebro) [inline sqlalchemy_drill/jdbc.py:94]: Missing line change.

File: test/test_jdbc.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        print(row)
+
+
+print("Now JDBC")
+jdbc_engine = create_engine('drill+jdbc://admin:password@localhost:31010')
+
+with jdbc_engine.connect() as con:
+    rs = con.execute('SELECT * FROM cp.`employee.json` LIMIT 5')
+    for row in rs:
+        print(row)
```
Comment (villebro) [inline test/test_jdbc.py]: Missing line change.

**Review by villebro — COMMENTED** [review]
I know this is WIP, just thought I'd drop a few comments that came to mind when browsing this PR.

**Comment by cgivre** [comment]
HI @JohnOmernik , 
Could you please take a look at this PR and if you like it, merge it?  Regarding the JDBC driver, I did some digging and decided that the best way to handle this was to have the user set an environment variable with the driver path and name.  

You theoretically could have `setup.py` d/l the driver, but I felt that doing so would cause a lot of problems.  Also, from what I was reading, it is generally not recommended.

**Pull request merged**